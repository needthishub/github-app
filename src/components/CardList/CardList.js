import React, {Component} from 'react';
import {Container} from './styledComponents';
import Card from '../Card/Card';

class CardList extends Component {
  render() {
    const {items} = this.props;
    return (
      <Container>
        {items.map((item) => (
          <Card
            key={item.id}
            link={item.html_url}
            title={item.name}
            description={item.description}
            image={item.owner.avatar_url}
          />
        ))}
      </Container>
    );
  }
}

export default CardList;
