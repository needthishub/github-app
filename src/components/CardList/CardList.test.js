import React from 'react';
import {shallow} from 'enzyme';
import CardList from './CardList';

describe('<CardList/>', () => {
  it('should match to snapshot', () => {
    const wrapper = shallow(<CardList items={[{
      id: '1',
      html_url: 'testUrl',
      name: 'testName',
      owner: {avatar_url: 'testUrl'},
      description: 'testText',
    }]}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
