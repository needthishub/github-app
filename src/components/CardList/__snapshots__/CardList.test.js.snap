// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList/> should match to snapshot 1`] = `ShallowWrapper {}`;
